### DISCUSSION TITLE
Collection camelCase relationship does not work (and collection names should not change casing in REST API)

### DISCUSSION THREAD

**Discussion by thefrana** [payloadcms/payload#994] [Feature Requests & Ideas]
# Bug Report

When using collection with two or more words in camelCase, eg. `myExamples`, one cannot build a relationship field.

```ts
const Users: CollectionConfig = {
  slug: 'users',
  auth: true,
  admin: {
    useAsTitle: 'email',
  },
  access: {
    read: () => true,
  },
  fields: [
    // Email added by default
    // Add more fields as needed
    {
      type: 'relationship',
      name: 'myExample',
      relationTo: 'myExamples'
    }
  ],
};
```

```ts
const Examples: CollectionConfig = {
  slug: 'myExamples',
  admin: {
    useAsTitle: 'someField',
  },
  fields: [
    {
      name: 'someField',
      type: 'text',
    },
  ],
}
```

## Expected Behavior

It should work `myExamples`, one cannot build a relationship field.

## Current Behavior

The server crashed right after start.

```
TypeError: Cannot read properties of undefined (reading 'config')
    at relationship (/home/luky/Projects/payload-relationship-repro/node_modules/payload/src/graphql/schema/buildMutationInputType.ts:148:68)
    at /home/luky/Projects/payload-relationship-repro/node_modules/payload/src/graphql/schema/buildMutationInputType.ts:214:12
    at Array.reduce (<anonymous>)
    at buildMutationInputType (/home/luky/Projects/payload-relationship-repro/node_modules/payload/src/graphql/schema/buildMutationInputType.ts:205:20)
    at /home/luky/Projects/payload-relationship-repro/node_modules/payload/src/collections/graphql/init.ts:131:85
    at Array.forEach (<anonymous>)
    at initCollectionsGraphQL (/home/luky/Projects/payload-relationship-repro/node_modules/payload/src/collections/graphql/init.ts:38:36)
    at registerSchema (/home/luky/Projects/payload-relationship-repro/node_modules/payload/src/graphql/registerSchema.ts:36:18)
    at init (/home/luky/Projects/payload-relationship-repro/node_modules/payload/src/init.ts:75:19)
    at initSync (/home/luky/Projects/payload-relationship-repro/node_modules/payload/src/init.ts:150:7)
[nodemon] app crashed - waiting for file changes before starting...
```

### Do not change collecation name casing when accessing via REST.

**Currently, said `myExamples` would be transformed to `my-examples` in REST API. I think this is bad from a numerous reasons:**

1. Devs should not expect changes in collection names. It feels unnatural.
2. The name is not changed constantly. Only when `/api/my-examples`. But when your query your field, the casing remains in camelCase. Also when accessing globals, it is not transformed: `/api/globals/myOtherGlobal`.
3. It creates unnecessary overhead. It could be easily skipped. Also, it would prevent point 1.

## Steps to Reproduce

https://github.com/lksfrn/payload-relationship-repro

## Detailed Description

Payload version: 1.0.27

**Comment by jmikrut** [comment]
Hey @lksfrn — this is a good catch. We are actually working on _**enforcing**_ snake-case for best-practice REST API reasons specifically. URL paths should always be snake-case.

We are going to improve our config validation so that it is impossible to write a camel-case collection slug, which will improve the DX here a bit. 

What do you think? I'm going to convert this to a Feature Request discussion, but I'd love to keep the conversation going there!

**Reply by thefrana** [reply]
Thanks for the reply. I thought about it. On one hand we have snake-case convention in REST APIs. However, would you rather write `/api/my-collection?where[my-prop][equals]=value` or `/api/my-collection?where[myProp][equals]=value`? The one need to translate props to snake-case everytime you need them and it brings another complexity and mental overhead. With camelCase props, it is much closer to the JS world, but not REST compliant.

On the other hand, I use REST APIs when I don't need complexity and the results are only one fetch away. For me, it is much more convenient to use camelCase everywhere: the database, the server code, the REST, the GraphQL and the client can all use it. 

Probably the best way to solve it is to create config option to disable snake-case conversion and use camelCase, but it will be enabled by default if you prefer snake-case.

**Comment by Rikhart** [comment]
I like to use a camelCase too, in  my case i want to map older collections it will be very helpfull to deactivate snake-case and use camelCase instead, another feature will be allow singular collection names.

**Reply by thefrana** [reply]
I ended up using just the lowercase letters for everything and I don't experience any issues
